Assets/Content/Items/Functional/Tools/WallConstructer.cs

```diff
@@ -32,6 +35,21 @@ public void Interact()
             var tileManager = FindObjectOfType<TileManager>();
 
             var tile = targetTile.Tile;
+
+            // The player using this item.
+            var player = transform.root;
+
+            // Cancel interaction if the target tile is outside the build range.
+            if (Vector3.Distance(player.transform.position, targetTile.transform.position) > buildDistance)
+            {
+                return;
+            }
+
+            if (tile.fixture != null) // Prevent construction if the tile is occupied by a fixture. 
+            {
+                return;
+            }
```

Comment:
Same as with TableConstructor, these two cases where nothing happens should be put in `CanInteract` (returning false if either of these two are true)